### ISSUE TITLE
Can you provide statistics for four submission results? 

### ISSUE THREAD

**Issue by chester-w-xie** [ilyassmoummad/dcase23_task5_scl#1]
Can you provide statistics for four submission results? I ran your code and obtained the statistics for the first submission result as follows. Could you confirm if they are correct?
Evaluation for: TESTteam VAL
BUK1_20181011_001004.wav {'TP': 11, 'FP': 4, 'FN': 20, 'total_n_pos_events': 31}
BUK4_20161011_000804.wav {'TP': 1, 'FP': 4, 'FN': 46, 'total_n_pos_events': 47}
BUK4_20171022_004304a.wav {'TP': 8, 'FP': 6, 'FN': 9, 'total_n_pos_events': 17}
BUK5_20161101_002104a.wav {'TP': 33, 'FP': 7, 'FN': 55, 'total_n_pos_events': 88}
BUK5_20180921_015906a.wav {'TP': 6, 'FP': 12, 'FN': 17, 'total_n_pos_events': 23}
ME1.wav {'TP': 10, 'FP': 34, 'FN': 1, 'total_n_pos_events': 11}
ME2.wav {'TP': 18, 'FP': 7, 'FN': 23, 'total_n_pos_events': 41}
R4_cleaned recording_13-10-17.wav {'TP': 19, 'FP': 0, 'FN': 0, 'total_n_pos_events': 19}
R4_cleaned recording_16-10-17.wav {'TP': 27, 'FP': 4, 'FN': 3, 'total_n_pos_events': 30}
R4_cleaned recording_17-10-17.wav {'TP': 36, 'FP': 0, 'FN': 0, 'total_n_pos_events': 36}
R4_cleaned recording_TEL_19-10-17.wav {'TP': 51, 'FP': 3, 'FN': 3, 'total_n_pos_events': 54}
R4_cleaned recording_TEL_20-10-17.wav {'TP': 63, 'FP': 0, 'FN': 1, 'total_n_pos_events': 64}
R4_cleaned recording_TEL_23-10-17.wav {'TP': 75, 'FP': 12, 'FN': 9, 'total_n_pos_events': 84}
R4_cleaned recording_TEL_24-10-17.wav {'TP': 99, 'FP': 0, 'FN': 0, 'total_n_pos_events': 99}
R4_cleaned recording_TEL_25-10-17.wav {'TP': 82, 'FP': 50, 'FN': 17, 'total_n_pos_events': 99}
file_423_487.wav {'TP': 57, 'FP': 2, 'FN': 0, 'total_n_pos_events': 57}
file_97_113.wav {'TP': 28, 'FP': 67, 'FN': 92, 'total_n_pos_events': 120}

Overall_scores: {'precision': 0.568118228922846, 'recall': 0.42415006534281063, 'fmeasure (percentage)': 48.569}

If you have the statistics for the other three submission results, please share them, and I'll be happy to assist you further.

Thanks for your code and nice work!

**Comment by ilyassmoummad** [comment]
Dear Wei Xie,

Thank you for reaching out and being interested in the work. I don't have the statistics for the validation set, but I performed many runs and picked the best checkpoint model to make the predictions for the test set. Which of the four system is this so I can look into it ?

The recall you obtained seems low, though I have to point out that training linear classifier with SGD for fine-tuning is very noisy for the audio files with more negative segments than positives (the results varied a lot from one run to another), probably a better sampling strategy to make sure the model doesn't update its parameters when it sees only negative samples but both positives and negatives would improve the results (which has not been explored in this work).

Thank you for the support !

Ilyass

**Comment by ilyassmoummad** [comment]
Update ! As I mentioned in my last comment, the inference method using cross-entropy is unstable. We have improved it (along with the pre-training loss) in our new work accepted at ICASSP 2024 where we obtain an F-score of 68% (on average and not best value as in this work) :

preprint : https://arxiv.org/abs/2309.08971
code: https://github.com/ilyassmoummad/RCL_FS_BSED

I hope this helps you !
